modules/programs/swappy.nix

```
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.swappy;
+
+  iniFormat = pkgs.formats.ini { };
+
+  iniFile = iniFormat.generate "config" cfg.settings;
+
+in {
+  meta.maintainers = [ hm.maintainers.eclairevoyant ];
+
+  options.programs.swappy = {
+    enable = mkEnableOption "Swappy";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.swappy;
+      defaultText = literalExpression "pkgs.swappy";
+      description = "Package providing <command>swappy</command>.";
+    };
+
+    settings = mkOption {
+      type = iniFormat.type;
+      default = {
+        Default = {
+          save_dir = "$HOME/Desktop";
+          save_filename_format = "swappy-%Y%m%d-%H%M%S.png";
+          show_panel = false;
+          line_size = 5;
+          text_size = 20;
+          text_font = "sans-serif";
+          paint_mode = "brush";
+          early_exit = false;
+          fill_shape = false;
+        };
+      };
```

Is there a strong reason why the default is not `{ }`? I think it would be best to have it empty to use the true Swappy defaults. For example, hopefully the true default would handle `save_dir = "$HOME/Desktop"` better for people who don't actually have that directory.

---

Yeah, I had a look and this does seem to be the case. See https://github.com/jtheoof/swappy/blob/596b9a8b17234303a9d4d2ea1d3107cb5442253e/src/config.c#L27

So I would strongly suggest trimming the default value to

```suggestion
      default = { };
```

---

Thanks, I made an incorrect assumption about the behavior of the app with no config provided. I tested with no config and it works as expected.